[CI] MixedClusterClientYamlTestSuiteIT test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates} failing

**Build Scans:**
- [elasticsearch-intake #10157 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lnot53iegwnf6)
- [elasticsearch-pull-request #32900 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/74zj7pe24dxyg)
- [elasticsearch-pull-request #32899 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ksx7b734whfv4)
- [elasticsearch-pull-request #32897 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vaf33irpo2tg6)
- [elasticsearch-intake #10155 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ajqyrywptbuui)
- [elasticsearch-pull-request #32889 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/wshdt34bwqy7e)
- [elasticsearch-pull-request #32891 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2we3i6sqteg4s)
- [elasticsearch-pull-request #32890 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/fqjyv3aqfq4hu)
- [elasticsearch-pull-request #32887 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vzwn3t7tssjqo)
- [elasticsearch-pull-request #32884 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/7jyvo2qgcesku)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.16.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}" -Dtests.seed=684135043AA70DF3 -Dtests.bwc=true -Dtests.locale=kl -Dtests.timezone=Etc/GMT+11 -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Dsearch%2F180_locale_dependent_mapping%2FTest%20Index%20and%20Search%20locale%20dependent%20mappings%20%2F%20dates%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search/180_locale_dependent_mapping:5]: got unexpected warning header [
	299 Elasticsearch-8.16.0-a2188010b76932ed1a86974715987b78ee2027ef "Date format [E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23. For more information, see https://ela.st/jdk-23-locales"
]

```

**Issue Reasons:**
- [main] 9 consecutive failures in test test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}
- [main] 9 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 7 consecutive failures in pipeline elasticsearch-pull-request
- [main] 46 failures in test test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates} (10.1% fail rate in 457 executions)
- [main] 44 failures in step 8.16.0_bwc-snapshots (11.0% fail rate in 399 executions)
- [main] 2 failures in step 8.16.0_bwc (6.7% fail rate in 30 executions)
- [main] 7 failures in pipeline elasticsearch-intake (5.5% fail rate in 128 executions)
- [main] 36 failures in pipeline elasticsearch-pull-request (12.8% fail rate in 282 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -284,9 +284,6 @@ tests:
 - class: org.elasticsearch.validation.DotPrefixClientYamlTestSuiteIT
   method: test {p0=dot_prefix/10_basic/Deprecated index template with a dot prefix index pattern}
   issue: https://github.com/elastic/elasticsearch/issues/113529
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search/180_locale_dependent_mapping/Test Index and Search locale dependent mappings / dates}
-  issue: https://github.com/elastic/elasticsearch/issues/113537
 - class: org.elasticsearch.xpack.inference.TextEmbeddingCrudIT
   method: testPutE5WithTrainedModelAndInference
   issue: https://github.com/elastic/elasticsearch/issues/113565

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
@@ -1,10 +1,10 @@
 ---
 "Test Index and Search locale dependent mappings / dates":
   - requires:
-      test_runner_features: ["allowed_warnings"]
+      test_runner_features: ["allowed_warnings-regex"]
   - do:
       allowed_warnings:
-        - "Date format [E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23"
+        - "Date format \\[E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23.*"
       indices.create:
           index: test_index
           body:

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml
@@ -1,9 +1,9 @@
 ---
 "Test Index and Search locale dependent mappings / dates":
   - requires:
-      test_runner_features: ["allowed_warnings-regex"]
+      test_runner_features: ["allowed_warnings_regex"]
   - do:
-      allowed_warnings:
+      allowed_warnings_regex:
         - "Date format \\[E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23.*"
       indices.create:
           index: test_index

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
@@ -554,7 +554,7 @@ emp_no:integer | year:long    | month:long    | day:long
 dateFormatLocale
 from employees | where emp_no == 10049 or emp_no == 10050 | sort emp_no 
 | eval birth_month = date_format("MMMM", birth_date) | keep emp_no, birth_date, birth_month;
-warningRegex:Date format \[MMMM\] contains textual field specifiers that could change in JDK 23
+warningRegex:Date format \[MMMM\] contains textual field specifiers that could change in JDK 23.*
 ignoreOrder:true
 
 emp_no:integer  |  birth_date:datetime       | birth_month:keyword

